iceoryx_hoofs/test/moduletests/test_posix_thread.cpp

```diff
@@ -35,64 +34,75 @@ class Thread_test : public Test
 
     void SetUp()
     {
-        m_run = true;
-        m_thread = new std::thread(&Thread_test::threadFunc, this);
     }
 
     void TearDown()
     {
-        m_run = false;
-        m_thread->join();
-        delete m_thread;
     }
 
     ~Thread_test()
     {
     }
 
-    void threadFunc()
-    {
-        while (m_run)
-        {
-            std::this_thread::sleep_for(std::chrono::milliseconds(10));
-        }
-    };
-
-    std::atomic_bool m_run{true};
-    std::thread* m_thread;
+    optional<Thread> sut;
 };
 
-#if !defined(__APPLE__)
+#if !defined(_WIN32) && !defined(__APPLE__)
```

Comment:
I think only the set and get name does not work on apple but the thread itself should be working. Could you run those tests also on apple?

Reply:
I didn't implement create and join yet. This will be done in the next PR.